GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -143,6 +147,37 @@ protected EventMetadata CreateEventMetadata(Exception e = null)
             return metadata;
         }
 
+        protected bool EnoughTimeBetweenRuns()
+        {
+            if (!this.Context.FileSystem.FileExists(this.TimeFilePath))
+            {
+                return true;
+            }
+
+            string lastRunTime = this.Context.FileSystem.ReadAllText(this.TimeFilePath);

```

Comment:
How are IO errors handled?

Reply:
I just pushed some changes that address this.  Great point!